src/UI/Implementation/Component/Input/Field/FilterContextRenderer.php

```diff
@@ -0,0 +1,240 @@
+
+		if ($component instanceof Component\Input\Field\Group) {
+			/**
+			 * @var $component Group
+			 */
+			return $this->renderFieldGroups($component, $default_renderer);
+		}
+
+		return $this->renderNoneGroupInput($component, $default_renderer);
+	}
+
+
+	/**
+	 * @param Component\Input\Field\Input $input
+	 * @param RendererInterface $default_renderer
+	 *
+	 * @return string
+	 */
+	protected function renderNoneGroupInput(Component\Input\Field\Input $input, RendererInterface $default_renderer) {
+		$input_tpl = null;
+
+		if ($input instanceof Component\Input\Field\Text) {
+			$input_tpl = $this->getTemplate("tpl.text.html", true, true);
+		} elseif ($input instanceof Component\Input\Field\Numeric) {
+			$input_tpl = $this->getTemplate("tpl.numeric.html", true, true);
+		} else {
+			throw new \LogicException("Cannot render '" . get_class($input) . "'");
+		}
+
+		return $this->renderProxyFieldWithContext($input_tpl, $input, $default_renderer);
+	}
+
+
+	/**
+	 * @param Group             $group
+	 * @param RendererInterface $default_renderer
+	 *
+	 * @return string
+	 */
+	protected function renderFieldGroups(Group $group, RendererInterface $default_renderer) {
+
+		$inputs = "";
+		foreach ($group->getInputs() as $input) {
+			$inputs .= $default_renderer->render($input);
+		}
+
+		$inputs .= $this->renderAddField($default_renderer);
+
+		return $inputs;
+	}
+
+
+	/**
+	 * @param Template $input_tpl
+	 * @param Input    $input
+	 * @param RendererInterface $default_renderer
+	 *
+	 * @return string
+	 */
+	protected function renderProxyFieldWithContext(Template $input_tpl, Input $input, RendererInterface $default_renderer) {
+
+		$f = $this->getUIFactory();
+		$tpl = $this->getTemplate("tpl.context_filter.html", true, true);
+
+		$tpl->setCurrentBlock("addon_left");
+		$tpl->setVariable("LABEL", $input->getLabel());
+		$tpl->parseCurrentBlock();
+		$tpl->setCurrentBlock("filter_field");
+		$tpl->setVariable("FILTER_FIELD", $this->renderProxyField($input_tpl, $input, $default_renderer));
+		$tpl->parseCurrentBlock();
+		$tpl->setCurrentBlock("addon_right");
+		$tpl->setVariable("DELETE", $default_renderer->render($f->glyph()->remove()));
+		$tpl->parseCurrentBlock();
+
+		return $tpl->get();
+	}
+
+
+	/**
+	 * @param Template $tpl
+	 * @param Input    $input
+	 * @param RendererInterface    $default_renderer
+	 *
+	 * @return string
+	 */
+	protected function renderProxyField(Template $input_tpl, Input $input, RendererInterface $default_renderer) {
+
+		$f = $this->getUIFactory();
+		$tpl = $this->getTemplate("tpl.filter_field.html", true, true);
+
+		$content = $this->renderInputFieldWithContext($input_tpl, $input);
+		$popover = $f->popover()->standard($f->legacy($content))->withVerticalPosition();
+		$tpl->setVariable("POPOVER", $default_renderer->render($popover));
+
+		$prox = new ProxyFilterField();
+		$prox = $prox->withOnClick($popover->getShowSignal());
+
+		$this->maybeRenderId($prox, $tpl);
+		return $tpl->get();
+	}
```

Comment:
Here is the proxy field again... As you can see it internally attaches the popovers with the Input Fields to the Filter Fields.